File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/fixtures/typescript/callbacks.ts
class Router {
  primary!: () => void; secondary!: () => void;
  setPrimary(cb: () => void) { this.primary = cb; } // store_primary
  setSecondary(cb: () => void) { this.secondary = cb; } // store_secondary
  firePrimary() { this.primary(); } // call_primary
  fireSecondary() { this.secondary(); } // call_secondary
}
class Other {
  primary!: () => void;
  firePrimary() { this.primary(); } // call_other
}

File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/fixtures/csharp/callbacks.cs
using System;
class Router {
    Action primary, secondary;
    void SetPrimary(Action cb) { this.primary = cb; } // store_primary
    void SetSecondary(Action cb) { this.secondary = cb; } // store_secondary
    void FirePrimary() { primary(); } // call_primary
    void FireSecondary() { secondary(); } // call_secondary
}
class Other {
    Action primary;
    void FirePrimary() { primary(); } // call_other
}
